server.js

```diff
@@ -1,33 +1,144 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixData from './data/netflix-titles.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
-//
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+
+// == Defining routes ==
+
+// list all the endpoints (_ works like a placeholder for req)
+app.get('/', (_, res) => {
+  res.send(listEndpoints(app))
+})
+
+// ** Endpoint to get all the shows **
+
+app.get('/shows', (req, res) => {
+  const { title, cast, country, director } = req.query
+  let shows = netflixData
+
+  // Filters so the user can search for items in the endpoint
+  if (title) {
+    shows = shows.filter(show => show.title.toString().includes(title))
+  }
+  if (cast) {
+    shows = shows.filter(show => show.cast.toString().includes(cast))
+  }
+  if (country) {
+    shows = shows.filter(show => show.country.toString().includes(country))
+  }
+  if (director) {
+    shows = shows.filter(show => show.director.toString().includes(director))
+  }
+
+  // Show an error message if the returning array is empty (no search result)
+  if (shows.length === 0) {
+    res.status(404).send('Not found')
+
+  // Show all data
+  } else {
+    res.json(shows)
+  }
+})
+
+// ** Endpoint to get only movies **
+
+app.get('/movies', (req, res) => {
+  let movies = netflixData.filter(show => show.type === "Movie")
+  const { title, cast, country, director } = req.query
+
+  // Filters so the user can search for items in the movies endpoint
+  if (title) {
+    movies = movies.filter(show => show.title.toString().includes(title))
+  }
+  if (cast) {
+    movies = movies.filter(show => show.cast.toString().includes(cast))
+  }
+  if (country) {
+    movies = movies.filter(show => show.country.toString().includes(country))
+  }
+  if (director) {
+    movies = movies.filter(show => show.director.toString().includes(director))
+  }
+
+  // Show an error message if the returning array is empty (no search result)
```

Comment:
You repeated this code, so it would be nice to extract it to a separate function and call it when needed 🤓